packages/devtools_app/lib/src/charts/chart_controller.dart

```diff
@@ -0,0 +1,449 @@
+// Copyright 2020 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/foundation.dart';
+import 'package:flutter/material.dart';
+import 'package:intl/intl.dart' as intl;
+
+import '../auto_dispose.dart';
+
+import 'chart_trace.dart';
+
+/// Zoom level of chart
+enum displayDuration { live, fiveMinutes, fifteenMinutes, all }
+
+/// Given a durationLabel return the displayDuration.
+displayDuration durationToDisplayDuration(
+  displayDuration currentZoomDuration,
+  Duration duration,
+) {
+  if (currentZoomDuration == displayDuration.live) {
+    if (duration.inMinutes == 0 && duration.inSeconds >= 10) {
+      return displayDuration.live;
+    }
+  } else if (currentZoomDuration == displayDuration.fifteenMinutes) {
+    if (duration.inMinutes >= 5) {
+      return displayDuration.fifteenMinutes;
+    }
+  } else if (currentZoomDuration == displayDuration.fiveMinutes) {
+    if (duration.inMinutes == 3) {
+      return displayDuration.fiveMinutes;
+    }
+  }
+
+  return null;
+}
+
+String prettyTimestamp(int timestamp) {
```

Comment:
not chart specific. Move this with our other time based helper utils.
@kenzieschmoll  we should also align with how we display times in the timeline page.